t/nqp/111-spawnprocasync.t

```diff
@@ -37,32 +42,83 @@ my $config := nqp::hash(
             $read_all := 1;
         }
     },
+    'stderr_bytes', -> $seq, $data, $err {
+        if nqp::isconcrete($data) {
+            @stderr_bytes[$seq] := $data;
+        }
+        else {
+            $read_all := 1;
+        }
+    },
     'buf_type', create_buf(uint8)
 );
 
-
+# define a task
+$string  := "expected output";
+$command := "echo $string";
+$args := $is-windows ?? nqp::list(nqp::getenvhash()<ComSpec>, '/c', $command)
+                     !! nqp::list('/bin/sh', '-c', $command);
 my $task := nqp::spawnprocasync($queue, $args, nqp::cwd(), nqp::getenvhash(), $config);
-
 nqp::permit($task, 1, -1);
 
+# run the task
 while !$done || !$read_all {
   if nqp::shift($queue) -> $task {
     if nqp::list($task) {
       my $code := nqp::shift($task);
       $code(|$task);
-    } else {
+    }
+    else {
       $task();
     }
   }
 }
 
-my class VMDecoder is repr('Decoder') {}
-my $dec := nqp::create(VMDecoder);
-nqp::decoderconfigure($dec, 'utf8', nqp::hash());
-
+# check results
 is($called_ready, 1, 'called the ready callback once');
-
 for @stdout_bytes -> $bytes {
     nqp::decoderaddbytes($dec, $bytes);
 }
-ok(nqp::decodertakeallchars($dec) ~~ /^aardvarks\s*$/, 'got the correct output');
+$s := nqp::decodertakeallchars($dec);
+ok($s ~~ / $string /, 'got the correct output on stdout');
+
+#== test for stderr ============
+# reset some reusable vars
+$done         := 0;
+$read_all     := 0;
+$called_ready := 0;
+
+# define the task
+$string  := "expected output";
+$command := "echo $string >&2";
+$args := $is-windows ?? nqp::list(nqp::getenvhash()<ComSpec>, '/c', $command)
+                     !! nqp::list('/bin/sh', '-c', $command);
+$task := nqp::spawnprocasync($queue, $args, nqp::cwd(), nqp::getenvhash(), $config);
+nqp::permit($task, 2, -1);
+
+# run the task
+while !$done || !$read_all {
+  if nqp::shift($queue) -> $task {
```

Comment:
This block seems to have 2 space indentation while everything else is using 4.

Reply:
Ok, will fix.

Reply:
Changes made and committed, all tests passing.